### PR TITLE
Class metadata fix for Windows

### PR DESCRIPTION
**Pull request by slavapestov**
On Windows we can't reference the runtime's Builtin.NativeObject value witness table directly from class metadata. Instead fill it in at runtime for singleton metadata initialization.

Do this on all platforms, even though its not needed on Linux or Darwin, for simplicity. A subsequent Windows-specific change will *force* the use of singleton metadata initialization for all classes, even when the class is otherwise fixed.

### CODE REVIEW

**Review by rjmccall — APPROVED** [review]
Generally LGTM, but some minor requests.

File: lib/IRGen/GenMeta.cpp [hunk 1]
```diff
@@ -2691,6 +2688,11 @@ namespace {
       B.addInt(IGM.SizeTy, 0);
     }
 
+    void addValueWitnessTable() {
+      // The runtime fills in the value witness table for us.
+      B.addInt(IGM.SizeTy, 0);
```
Comment (compnerd) [inline lib/IRGen/GenMeta.cpp]: Why not `IGM.ValueWitnessTableTy`?

Comment (slavapestov) [reply lib/IRGen/GenMeta.cpp]: There doesn't appear to be such a thing but elsewhere we use IGM.WitnessTablePtrTy. I'll use that.

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

File: lib/IRGen/GenMeta.cpp (hunk 1 above)
Comment (rjmccall) [reply lib/IRGen/GenMeta.cpp]: Seems reasonable to add as an alias just for self-documenting interests.

**Pull request merged**